src/charm.py

```diff
@@ -121,24 +70,116 @@ def _on_update_status(self, event):
         - Unit is active
         """
 
-        if not self.unit.is_leader():
-            self.unit.status = ActiveStatus()
-            return
-
         if not self.mysql.is_ready():
             status_message = "MySQL not ready yet"
             self.unit.status = WaitingStatus(status_message)
             return
 
-        if not self._stored.mysql_initialized:
+        if not self._is_mysql_initialized():
             status_message = "MySQL not initialized"
             self.unit.status = WaitingStatus(status_message)
             return
 
         self.unit.status = ActiveStatus()
 
+    ##############################################
+    #               PROPERTIES                   #
+    ##############################################
+    @property
+    def mysql(self) -> MySQL:
+        """Returns MySQL object"""
+        peers_data = self.model.get_relation("mysql").data[self.app]
+        mysql_config = {
+            "app_name": self.model.app.name,
+            "host": self.unit_ip,
+            "port": self.model.config["port"],
+            "user_name": "root",
+            "mysql_root_password": peers_data["mysql_root_password"],
+        }
+        return MySQL(mysql_config)
+
+    @property
+    def unit_ip(self) -> str:
+        """Returns unit's IP"""
+        return str(self.model.get_binding(PEER).network.bind_address)
+
+    @property
+    def env_config(self) -> dict:
+        """Return the env_config for pebble layer"""
+        config = self.model.config
+        peers_data = self.model.get_relation("mysql").data[self.app]
+        env_config = {}
+        env_config["MYSQL_ROOT_PASSWORD"] = peers_data["mysql_root_password"]
+
+        if config.get("MYSQL_USER") and config.get("MYSQL_PASSWORD"):
+            env_config["MYSQL_USER"] = config["MYSQL_USER"]
+            env_config["MYSQL_PASSWORD"] = config["MYSQL_PASSWORD"]
```

Comment:
To have a minor reduction on code dup you may want to consider the walrus:
```python
if (user := config.get("MYSQL_USER")) and (password := config.get("MYSQL_PASSWORD")):
    env_config["MYSQL_USER"] = user
    env_config["MYSQL_PASSWORD"] = password
```

Reply:
I am the warlus!